cmd/minishift/cmd/openshift/component_list.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openshift
+
+import (
+	"fmt"
+	minishiftConstants "github.com/minishift/minishift/pkg/minishift/constants"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+// version command represent current running openshift version and available one.
+var componentListCmd = &cobra.Command{
+	Use:   "list [component-name]",
+	Short: "Add component to an OpenShift cluster (Only works openshift version >= 3.10.x)",
+	Long:  `Add component to an OpenShift cluster (Only works openshift version >= 3.10.x)`,
+	Run:   runComponentList,
+}
+
+func runComponentList(cmd *cobra.Command, args []string) {
+	fmt.Fprint(os.Stdout, "The following OpenShift components are available: \n")
+	for _, component := range minishiftConstants.ValidComponents {
+		fmt.Fprintf(os.Stdout, "\t- %s\n", component)
```

Comment:
We need to repeat that these are OpenShift components as the command line says it explicitly. Suggestion:

```diff
func runComponentList(cmd *cobra.Command, args []string) {
-       fmt.Fprint(os.Stdout, "The following OpenShift components are available: \n")
+       fmt.Fprint(os.Stdout, "Available components: \n")
        for _, component := range minishiftConstants.ValidComponents {
-               fmt.Fprintf(os.Stdout, "\t- %s\n", component)
+               fmt.Fprintf(os.Stdout, "- %s\n", component)
        }
 }
```
Output
```
$ minishift openshift component list
Available components:
- automation-service-broker
- service-catalog
- template-service-broker
```

Reply:
@LalatenduMohanty I did it because we want to be consistent with `minishift openshift version list`, don't want to break this.

```
$ ./minishift openshift version list
The following OpenShift versions are available: 
	- v3.9.0
	- v3.10.0
```

Reply:
@praveenkumar I still think we should change it to `fmt.Fprint(os.Stdout, "Available components: \n")`. I feel the current output print a long statement which is not required.

Reply:
I can fix this during the merge too.

Reply:
@LalatenduMohanty one way we are thinking to have same kind of output for different commands and then we are changing here. I don't have any strong option here but doesn't seem right.